tests/utils_tests/test_http.py

```diff
@@ -165,6 +165,32 @@ def test_no_allowed_hosts(self):
         # Basic auth without host is not allowed.
         self.assertIs(is_safe_url(r'http://testserver\@example.com'), False)
 
+    def test_is_safe_url_from_settings(self):
+        test_cases = [
+            ('https://testserver/', ['testserver', 'otherserver'], True),
+            ('https://testserver/', ['*'], True),
+            ('https://testserver/', {'*'}, True),
+            ('https://testserver/', ['otherserver', '*'], True),
+            ('https://testserver/', '*', True),
+            ('https://testserver/', [], False),
+            ('https://testserver/', {}, False),
+            ('https://testserver/', ['otherserver'], False),
+            ('https://testserver/', {'otherserver'}, False),
+            ('https://otherserver/', ['testserver', 'otherserver'], True),
+            ('https://otherserver/', ['*'], True),
+            ('https://otherserver/', {'*'}, True),
+            ('https://otherserver/', ['testserver', '*'], True),
+            ('https://otherserver/', '*', True),
+            ('https://otherserver/', [], False),
+            ('https://otherserver/', {}, False),
+            ('https://otherserver/', ['testserver'], False),
+            ('https://otherserver/', {'testserver'}, False),
+        ]
+        for url, allowed_hosts, expected_answer in test_cases:
+            with override_settings(ALLOWED_HOSTS=allowed_hosts), self.subTest(url=url, allowed_hosts=allowed_hosts,
+                                                                              expected_answer=expected_answer):
+                self.assertEqual(is_safe_url(url), expected_answer)
```

Comment:
For verifying booleans, `assertIs` is preferred. From the [unittest docs](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertTrue)

> Note that [`assertTrue`] is equivalent to bool(expr) is True and not to expr is True (use assertIs(expr, True) for the latter).

```py
self.assertIs(is_safe_url(url), expected_answer)
```

In other words, `is_safe_url()` is expected to return the object `True`, not just something equal to `True`.